include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -581,7 +597,10 @@ struct __get_sycl_range
         assert(__first < __last);
         using value_type = val_t<_Iter>;
 
-        const auto __offset = __first - oneapi::dpl::begin(__first.get_buffer());
+        // Types for which oneapi::dpl::__ranges::is_sycl_iterator_v = true should have both:
```

Comment:
There is no explanation in the comments why we need new code with `get_idx()`.

Reply:
comment added.